src/topology/stone_cech.lean

```diff
@@ -307,4 +308,53 @@ end
 instance stone_cech.compact_space : compact_space (stone_cech α) :=
 quotient.compact_space
 
+section universal_property
+
+variables {δ : Type u} [topological_space δ]
+variables {δ₁ : Type u} [topological_space δ₁] [t2_space δ₁] [compact_space δ₁]
+variables {δ₂ : Type u} [topological_space δ₂] [t2_space δ₂] [compact_space δ₂]
+
+/-- The universal property that uniquely characterizes the Stone-Čech compactification of `α`, when
+  `stone_cech_unit` is an embedding. Notice that `stone_cech α` has this property
+  in general by `stone_cech_extend_extends`, `continuous_stone_cech_extend` and
+  `stone_cech_hom_ext`, but lemma `stone_cech_extension_property_unique` assumes a dense
+  embedding that `stone_cech_unit` is not guaranteed to be. -/
+def stone_cech_extension_property (e : α → δ) :=
+  ∀ {γ : Type u} [topological_space γ], by exactI ∀ [t2_space γ]
+  [compact_space γ] (f : α → γ) (hf: continuous f),
+  ∃! (ext_f : δ → γ), continuous ext_f ∧ f = ext_f ∘ e
```

Comment:
Can you add some API for this unique map. As you can see below, you need to do quite a lot of work to access it. It's best to do that work only once, by making a new definition
```
def stone_cech_extension_property.extend
```
that extracts this map (use `.some` to do this),
and then some lemmas that show that extends, that it is continuous, that it is unique.

That way, the proofs below will become much cleaner and shorter. And idem dito for future use.